core/src/main/java/org/infinispan/persistence/async/AsyncCacheWriter.java

```diff
@@ -144,6 +191,14 @@ public void stop() {
       }
    }
 
+   @Override
+   public boolean isAvailable() {
+      // Available if failSilently || actual == available || actual != available and queue has capacity
+      // Worst case, writeBatch comes in before !Available is declared by Async, in which case it will wait on
+      // writeLock until the stateLock is reset when the queue is finally flushed
+      return !stopped && (asyncConfiguration.failSilently() || delegateAvailable || stateLock.hasCapacity());
```

Comment:
It seems updating the `delegateAvailable` every time this is invoked would be remedy the issue of having another timer task?